command/agent/consul/connect.go

```diff
@@ -84,27 +88,38 @@ func newConnectGateway(serviceName string, connect *structs.ConsulConnect) *api.
 	return &api.AgentServiceConnectProxyConfig{Config: envoyConfig}
 }
 
-func connectSidecarRegistration(serviceName string, css *structs.ConsulSidecarService, networks structs.Networks) (*api.AgentServiceRegistration, error) {
+func connectSidecarRegistration(serviceId string, css *structs.ConsulSidecarService, networks structs.Networks, ports structs.AllocatedPorts) (*api.AgentServiceRegistration, error) {
 	if css == nil {
 		// no sidecar stanza means there is no sidecar service to register
 		return nil, nil
 	}
 
-	cNet, cPort, err := connectPort(serviceName, networks)
+	cMapping, err := connectPort(css.Port, networks, ports)
 	if err != nil {
 		return nil, err
 	}
 
-	proxy, err := connectSidecarProxy(css.Proxy, cPort.To, networks)
+	proxy, err := connectSidecarProxy(css.Proxy, cMapping.To, networks)
 	if err != nil {
 		return nil, err
 	}
 
 	return &api.AgentServiceRegistration{
 		Tags:    helper.CopySliceString(css.Tags),
-		Port:    cPort.Value,
-		Address: cNet.IP,
+		Port:    cMapping.Value,
+		Address: cMapping.HostIP,
 		Proxy:   proxy,
+		Checks: api.AgentServiceChecks{
+			{
+				Name:     "Connect Sidecar Listening",
+				TCP:      ipaddr.FormatAddressPort(cMapping.HostIP, cMapping.Value),
```

Comment:
Let's remove the extra non-`api/` consul dependency and just copy the implementation of `FormatAddressPort`, which is one line
```
func FormatAddressPort(address string, port int) string {                                                                                           
        return net.JoinHostPort(address, strconv.Itoa(port))                                                                                        
} 
```